src/plugins/typingIndicator.tsx

```diff
@@ -0,0 +1,281 @@
+ * Vencord, a modification for Discord's desktop app
+ * Copyright (c) 2022 Vendicated and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+import { Message } from "discord-types/general";
+
+import { addServerListElement, removeServerListElement, ServerListRenderPosition } from "../api/ServerList";
+import { Devs } from "../utils/constants";
+import { openPrivateChannel } from "../utils/discord";
+import definePlugin, { OptionType } from "../utils/types";
+import { Settings } from "../Vencord";
+import { ChannelStore, ContextMenu, FluxDispatcher, Forms, Menu, React, UserStore } from "../webpack/common";
+
+interface ITyping {
+    channelId: string;
+    userId: string;
+}
+
+interface IMessageCreate {
+    type: "MESSAGE_CREATE";
+    optimistic: boolean;
+    isPushNotification: boolean;
+    channelId: string;
+    message: Message;
+}
+
+const dotsIcon = () => (
+    <svg
+        className="circleIcon"
+        height="24"
+        width="24"
+        viewBox="0 0 24 24"
+        fill="currentColor"
+    >
+        <path
+            d="M12,1.5A1.5,1.5,0,1,1,13.5,3,1.5,1.5,0,0,1,12,1.5Zm-6,0A1.5,1.5,0,1,1,7.5,3,1.5,1.5,0,0,1,6,1.5Zm-6,0A1.5,1.5,0,1,1,1.5,3,1.5,1.5,0,0,1,0,1.5Z"
+            transform="translate(4.5 11)"
+        />
+    </svg>
+);
+
+const typingUsers: Array<ITyping> = [];
+let toolTipString: string = "";
+
+let buttonType: any;
+let buttonProps: any;
+let buttonChildren: any;
+
+function ContextMenuElement() {
+    const buttons: Array<JSX.Element> = [];
+
+    for (let i = 0; i < typingUsers.length; i++) {
+        const user = UserStore.getUser(typingUsers[i].userId);
+        if (user) {
+            buttons.push(
+                <Menu.MenuItem
+                    id={`typing-${i}`}
+                    label={user.username}
+                    action={() => {
+                        openPrivateChannel(user.id);
+                    }}
+                />
+            );
+        }
+    }
+
+    return <Menu.ContextMenu
+        navId="typing-indicator-menu"
+        onClose={() => FluxDispatcher.dispatch({ type: "CONTEXT_MENU_CLOSE" })}>
+        {buttons}
+    </Menu.ContextMenu>;
+}
+
+let forceRenderIndicator: React.DispatchWithoutAction;
+
+const addUser = (channelId: string, userId: string) => {
+
+    const user = typingUsers.find(o => o.channelId === channelId && o.userId === userId);
```

Comment:
```suggestion
    const hasUser = typingUsers.some(o => o.channelId === channelId && o.userId === userId);
```
But in any case, an array is not the right data structure here, use a Set which has add and remove operations